client/src/project/new/ProjectNew.container.js

```diff
@@ -277,27 +279,86 @@ function ForkProject(props) {
   );
 }
 
+/**
+ * Validate and decode query params.
+ *
+ * @param {object} params - query params
+ * @returns {object} parsed parameters, validated and ready to be be used to pre-fill fields.
+ */
+function getDataFromParams(params) {
+  if (params && Object.keys(params).length) {
+    if (params.data) {
+      const data = JSON.parse(atobUTF8(params.data));
+      if (data && Object.keys(data).length) {
+        // validate metadata
+        const validKeys = Object.keys(newProjectSchema.createEmpty().automated.data);
+        const keys = Object.keys(data);
+        for (let key of keys) {
+          if (!validKeys.includes(key))
+            throw new Error("unexpected project field in the encoded metadata: " + key);
+        }
+      }
+      return data;
+    }
+    // Unrecognized params: should we notify? Let's start without notifications.
+    return null;
+  }
```

Comment:
I suggest rewriting it with guarding clauses. I think this will be advantageous if we switch to notifications, since then we will know what the problem is.

```suggestion
    // Unrecognized params: should we notify? Let's start without notifications.
  const errorValue = null;
  if (params  == null) return errorValue;
  if (Object.keys(params).length < 1) return errorValue;
  if (params.data == null) return errorValue;
  const data = JSON.parse(atobUTF8(params.data));
  if (data == null || Object.keys(data).length < 1) return errorValue;
  // validate metadata
  const validKeys = Object.keys(newProjectSchema.createEmpty().automated.data);
  const keys = Object.keys(data);
  for (let key of keys) {
    if (!validKeys.includes(key))
      throw new Error("unexpected project field in the encoded metadata: " + key);
  }
  return data;
```

Reply:
The code you suggested is definitely less convoluted.
I ended up committing a small variation of it, using `|| !` to merge the conditions in the first few lines so that it's coherent with line `295`. Let me know if that works